### ISSUE TITLE
Replicate test structure used for `Install and enable` tests in GitHub Action `Test indexer`

### ISSUE THREAD

**Issue by verdx** [wazuh/wazuh-packages#2080]
|Wazuh version|Install type|Action performed|Platform|
|---|---|---|---|
| 4.5 | Tests | 

Right now, the [Test of the indexer](https://github.com/wazuh/wazuh-packages/actions/workflows/test-indexer-rpm.yml) creates the docker image and the package to test the indexer. The same structure should be used as in [Test install and enable](https://github.com/wazuh/wazuh-packages/actions/workflows/test-install-and-enable-deb.yml), where the docker image is created in a separate GitHub Action when needed, and the package is created one time in a different workflow and uploaded as an artifact. 

This way, more tests will be able to be done using the image and package created and they will only have to be created when needed.

**Comment by gdiazlo** [comment]
Blocked by https://github.com/wazuh/wazuh-indexer/issues/4

**Comment by AlexRuiz7** [comment]
Superseded by https://github.com/wazuh/wazuh-indexer/issues/174

Won't fix.
